src/qibocal/utils.py

```diff
@@ -1,13 +1,11 @@
-from typing import Dict, List, Union
+from typing import Dict, Optional
 
 from qibolab.platform import Platform
 from qibolab.qubits import Qubit, QubitId
 
 
 def allocate_qubits(
-    platform: Platform, qubit_ids: List[QubitId]
-) -> Union[List[QubitId], Dict[QubitId, Qubit]]:
+    platform: Optional[Platform], qubit_ids: list
```

Comment:
```suggestion
    platform: Optional[Platform], qubit_ids: List[QubitId]
```
Keep the type as specific as possible (unless you really want to allow for a whatever list to be passed).